twitter/View Controllers/ComposeViewController.m

```diff
@@ -0,0 +1,72 @@
+//
+//  ComposeViewController.m
+//  twitter
+//
+//  Created by Mai Ngo on 6/21/22.
+//  Copyright © 2022 Emerson Malca. All rights reserved.
+//
+
+#import "ComposeViewController.h"
+#import "APIManager.h"
+
+@interface ComposeViewController () <UITextViewDelegate>
+@property (weak, nonatomic) IBOutlet UITextView *textView;
+@property (weak, nonatomic) IBOutlet UILabel *characterCountLabel;
+@property (weak, nonatomic) IBOutlet UILabel *warningLabel;
+@property (weak, nonatomic) IBOutlet UIBarButtonItem *tweetButton;
+
+@end
+
+@implementation ComposeViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view.
+    self.textView.delegate = self;
+    [self.warningLabel setHidden:YES];
+
+}
+- (void)textViewDidChange:(UITextView *)textView {
+    NSString *text = self.textView.text;
+    NSInteger *count = [text length];
+    NSString *countString = [NSString stringWithFormat:@"%lu", count];
```

Comment:
This is personal preference, but pointing it out just to think about it--you don't necessarily need a new variable for every property you're accessing. I would include it if it makes things more readable, or if the line is too long, but getting the text length is a common enough operation that it doesn't really require extra explanation. So I would have made this just:

`NSString *const countString = [NSString stringWithFormat:@"%lu", self.textView.text.length];`

or even just:

`self.characterCountLabel.text = [NSString stringWithFormat:@"%lu", self.textView.text.length];`

is still pretty readable to me (possibly with a newline after the stringWithFormat string if you don't want the line to get too long).

Reply:
For this part, I created the `count` variable so that I could use it later to compared to the `maxCharAllowed` variable